### ISSUE TITLE
Unauthenticated graph access not working

### ISSUE THREAD

**Issue by mctaguma** [librenms/librenms#14263]
### The problem

When I try to access a graph without being authenticated, I see an error in logs/librenms.log, included below.

I have the following in config.php:

`$config['allow_unauth_graphs'] = 1;`

I am using PHP 7.3 due to utilising PHP Weathermap, which I need to improve prior to being able to use PHP 8.1.

### Output of ./validate.php

```txt
====================================
Component | Version
--------- | -------
LibreNMS  | 21.6.0-1181-g003f57032
DB Schema | 2022_07_03_1947_add_app_data (244)
PHP       | 7.3.33
Python    | 3.6.8
Database  | MariaDB 10.6.8-MariaDB
RRDTool   | 1.7.0
SNMP      | 5.8
====================================

[OK]    Composer Version: 2.4.1
[OK]    Dependencies up-to-date.
[OK]    Database connection successful
[OK]    Database Schema is current
[OK]    SQL Server meets minimum requirements
[OK]    lower_case_table_names is enabled
[OK]    MySQL engine is optimal
[OK]
[OK]    Database schema correct
[OK]    MySQl and PHP time match
[WARN]  PHP version 8.1 is the minimum supported version as of September, 2022. We recommend you update PHP to a supported version (8.1 suggested) to continue to receive updates. If you do not update PHP, LibreNMS will continue to function but stop receiving bug fixes and updates.
[OK]    Active pollers found
[OK]    Dispatcher Service not detected
[OK]    Locks are functional
[OK]    Python poller wrapper is polling
[OK]    Redis is unavailable
[OK]    rrdtool version ok
[OK]    Connected to rrdcached
```

### Anything in the logs that might be useful for us?

```txt
production.ERROR: Argument 2 passed to App\Models\DeviceRelatedModel::scopeHasAccess() must be an instance of App\Models\User, null given, called in /opt/librenms/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Model.php on line 1475 {"exception":"[object] (TypeError(code: 0): Argument 2 passed to App\\Models\\DeviceRelatedModel::scopeHasAccess() must be an instance of App\\Models\\User, null given, called in /opt/librenms/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Model.php on line 1475 at /opt/librenms/app/Models/DeviceRelatedModel.php:35)"}
```

**Comment by sauri1** [comment]
I have similar problem; unauthenticated graphs show only dark red text "multiport*bits_separate". (When authenticated, all graphs work also normally.)
Problem seems to be in commit 6f5cf7727a36e1bf0fca6be7779973fe82b4efad because last commit (797d73ad166b071576312ed0ca20200dce955889) works normally.

**Comment by kruczek8989** [comment]
i Hae similar problem too, it's stopped working from since 31.08

**Comment by sauri1** [comment]
Problem seems to be in commit 6f5cf7727a36e1bf0fca6be7779973fe82b4efad, file includes/html/graphs/graph.inc.php (https://github.com/librenms/librenms/commit/6f5cf7727a36e1bf0fca6be7779973fe82b4efad#diff-fb41139b6fb010b0811ccd73c7e4202f8fcb72a12b5c9076fe03a6a46be2a274):
`$auth = false;`
Commenting this line shows graph without being authenticated, as before this change.

**Comment by kruczek8989** [comment]
> Problem seems to be in commit [6f5cf77](https://github.com/librenms/librenms/commit/6f5cf7727a36e1bf0fca6be7779973fe82b4efad), file `includes/html/graphs/graph.inc.php` ([6f5cf77#diff-fb41139b6fb010b0811ccd73c7e4202f8fcb72a12b5c9076fe03a6a46be2a274](https://github.com/librenms/librenms/commit/6f5cf7727a36e1bf0fca6be7779973fe82b4efad#diff-fb41139b6fb010b0811ccd73c7e4202f8fcb72a12b5c9076fe03a6a46be2a274)): `$auth = false;` Removing this line shows graph without being authenticated, as before this change.

That's  resolving problem, graph now is working.

**Comment by murrant** [comment]
Fixed by #14268
